### PR TITLE
[YAML] Expose flatten implementation from Java.

### PR DESCRIPTION
**Pull request by robertwb**
Technically Flatten will produce exactly the same result no matter where it is expanded, but simply having this in Java avoids complexities around dealing with affinity optimization.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by robertwb** [comment]
R: @Polber 

I do not think the failures are related.

File: sdks/python/apache_beam/yaml/standard_providers.yaml [hunk 1]
```diff
@@ -24,6 +24,29 @@
...
+    version: BEAM_VERSION
+  transforms:
+     Flatten: 'beam:schematransform:org.apache.beam:yaml:flatten:v1'
+
+- type: 'beamJar'
+  config:
+    gradle_target: 'sdks:java:extensions:schemaio-expansion-service:shadowJar'
+    version: BEAM_VERSION
+  transforms:
+     Flatten: 'beam:schematransform:org.apache.beam:yaml:flatten:v1'
```
Comment (Polber) [inline sdks/python/apache_beam/yaml/standard_providers.yaml:49]: Why are there multiple gradle targets? If it is in multiple, then why the need to specify multiple?

Comment (robertwb) [reply sdks/python/apache_beam/yaml/standard_providers.yaml:49]: The idea here is that we want to be able to use the same environment as proceeding/subsequent operations when possible.

Comment (Polber) [reply sdks/python/apache_beam/yaml/standard_providers.yaml:49]: Ah ok that makes sense. In that case, should we do the same for other transforms? I assume similar providers (i.e. MapToFields, Filter, etc.) also get packaged in those targets?

Comment (robertwb) [reply sdks/python/apache_beam/yaml/standard_providers.yaml:49]: I'm thinking about how to do this automatically.

**Review by Polber — APPROVED** [review]

File: sdks/python/apache_beam/yaml/standard_providers.yaml (hunk 1 above)
Comment (Polber) [reply sdks/python/apache_beam/yaml/standard_providers.yaml:49]: LGTM in the meantime

**Pull request merged**